pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
+							},
+						},
+					},
+				},
+				ServiceName:         "basic-client",
+				PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+				UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilImagePullSecrets := func() []corev1.LocalObjectReference {
+		return []corev1.LocalObjectReference{
+			{
+				Name: "basic-secrets",
+			},
+		}
+	}
+
+	newNonNilTolerations := func() []corev1.Toleration {
+		return []corev1.Toleration{
+			{
+				Key:      "key",
+				Operator: corev1.TolerationOpEqual,
+				Value:    "value",
+				Effect:   corev1.TaintEffectNoSchedule,
+			},
+		}
+	}
+
+	newNonNilNodeAffinity := func() *corev1.NodeAffinity {
+		return &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "key",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"value"},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilPodAffinity := func() *corev1.PodAffinity {
+		return &corev1.PodAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilPodAntiAffinity := func() *corev1.PodAntiAffinity {
+		return &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	tt := []struct {
+		name                string
+		rack                scyllav1.RackSpec
+		scyllaCluster       *scyllav1.ScyllaCluster
+		existingStatefulSet *appsv1.StatefulSet
+		sidecarImage        string
+		expectedStatefulSet *appsv1.StatefulSet
+		expectedError       error
+	}{
+		{
+			name:                "new StatefulSet",
+			rack:                newBasicRack(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        "",
+			expectedStatefulSet: newBasicStatefulSet(),
+			expectedError:       nil,
+		},
+		{
+			name: "error for invalid Rack storage",
+			rack: func() scyllav1.RackSpec {
+				r := newBasicRack()
+				r.Storage = scyllav1.StorageSpec{
+					Capacity: "",
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        "",
+			expectedStatefulSet: nil,
+			expectedError:       fmt.Errorf(`cannot parse "": quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'`),
```

Comment:
given the error is public, you could use ```fmt.Errorf(`cannot parse "": %s`, ErrFormatWrong)``` or ```fmt.Errorf(`cannot parse "": %s`, errors.New("quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'"))```

Reply:
fixed